rest-service/manager_rest/upload_manager.py

```diff
@@ -614,10 +627,22 @@ def _prepare_and_process_doc(self,
                                  file_server_root,
                                  archive_target_path,
                                  **kwargs):
+
+        # support previous implementation
+        wagon_target_path = archive_target_path
+
+        # handle the archive_target_path, which may be zip or wagon
+        if zipfile.is_zipfile(archive_target_path):
+            archive_name = unzip(archive_target_path,
+                                 logger=current_app.logger)
+            os.remove(archive_target_path)
+            shutil.move(archive_name, archive_target_path)
+            wagon_target_path, _ = self._verify_archive(archive_target_path)
```

Comment:
add a comment about why the returned `yamls[0]` can be ignored?

Reply:
no special reason, I just don't use it.. :)